lib/python-sdk/common_grants_sdk/client/opportunity.py

```diff
@@ -1,38 +1,28 @@
 """Opportunity namespace for the CommonGrants API."""
 
 import httpx
-from typing import Callable
+from typing import TYPE_CHECKING
 from uuid import UUID
 
-from .auth import Auth
-from .config import Config
 from .exceptions import raise_api_error
 from ..schemas.pydantic.models import OpportunityBase
 from ..schemas.pydantic.responses import OpportunityResponse, OpportunitiesListResponse
 
 
+if TYPE_CHECKING:
+    from .client import Client
+
+
 class Opportunity:
     """Class for fetching opportunity data from CommonGrants API."""
 
-    def __init__(
-        self,
-        auth: Auth,
-        config: Config,
-        http: httpx.Client,
-        url: Callable[[str], str],
-    ):
+    def __init__(self, client: "Client"):
```

Comment:
This is still referencing the `Client` class, not a literal string, but it's just how Python allows you to avoid circular imports and errors at runtime.